hera_cal/abscal.py

```diff
@@ -981,510 +2636,319 @@ def TT_Phi_gain_arr(self):
+                        model, model_flags, _ = io.partial_time_io(hdm, model_times_to_load, polarizations=[pol])
+                        model_bls = {bl: model.antpos[bl[0]] - model.antpos[bl[1]] for bl in model.keys()}
+                        utils.lst_rephase(model, model_bls, model.freqs, data.lsts - model.lsts,
+                                          lat=hdm.telescope_location_lat_lon_alt_degrees[0], inplace=True)
+                        
+                        # update data flags w/ model flags
+                        for k in flags.keys():
+                            if k in model_flags:
+                                flags[k] += model_flags[k]
+
+                        # apply calibration
+                        data_ants = set([ant for bl in data.keys() for ant in split_bl(bl)])
+                        rc_gains_subset = {k: rc_gains[k][tinds, :] for k in data_ants}
+                        rc_flags_subset = {k: rc_flags[k][tinds, :] for k in data_ants}
+                        calibrate_in_place(data, rc_gains_subset, data_flags=flags, 
+                                           cal_flags=rc_flags_subset, gain_convention=hc.gain_convention)
+                        
+                        # run absolute calibration, copying data because it gets modified internally
+                        delta_gains, AC = post_redcal_abscal(model, copy.deepcopy(data), flags, rc_flags_subset, edge_cut=edge_cut, 
+                                                             tol=tol, min_bl_cut=min_bl_cut, max_bl_cut=max_bl_cut, 
+                                                             gain_convention=hc.gain_convention, phs_max_iter=phs_max_iter, 
+                                                             phs_conv_crit=phs_conv_crit, verbose=verbose)
+
+                        # abscal data (AC.data is already abscaled, but data is only redcaled) and generate abscal Chi^2
+                        calibrate_in_place(data, delta_gains, data_flags=flags, 
+                                           cal_flags=rc_flags_subset, gain_convention=hc.gain_convention)
+                        chisq_wgts = {}
+                        for bl in AC.data.keys():
+                            dt = (np.median(np.ediff1d(hd.times_by_bl[bl[:2]])) * 86400.)
+                            noise_var = predict_noise_variance_from_autos(bl, data, dt=dt)
+                            chisq_wgts[bl] = noise_var**-1 * (~flags[bl]).astype(np.float)
+                        total_qual, nObs, quals, nObs_per_ant = utils.chisq(AC.data, AC.model, chisq_wgts,
+                                                                            gain_flags=rc_flags_subset, split_by_antpol=True)
+                    
+                        # update results
+                        delta_flags = synthesize_ant_flags(flags)
+                        for ant in data_ants:
+                            abscal_gains[ant][tinds, :] = rc_gains_subset[ant] * delta_gains[ant]
```

Comment:
I think you're double counting `rc_gains` here b/c `abscal_gains` was a deepcopy to begin with.

Reply:
No, because I'm doing = rather than *=, so I'm overwriting.